### PR TITLE
minor details + non-recursive algo in integer_valued_polynomials

### PR DESCRIPTION
**Pull request by fchapoton**
This adds a better algo (non-recursive) for the shift of variables in integer-valued polynomial rings.

Also some little tweaks to the doc in this file.

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [ ] I have linked a relevant issue or discussion.
- [x] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/rings/polynomial/integer_valued_polynomials.py [hunk 1]
```diff
@@ -769,24 +774,23 @@ def variable_shift(self, k=1):
...
+                            for j in range(n + 1)}
 
-                resu = self - A._from_dict({i - 1: c for i, c in self if i})
-                if k == -1:
-                    return resu
-                return resu.variable_shift(k + 1)
+                from sage.data_structures.blas_dict import linear_combination
+                mc = self._monomial_coefficients
+                ret = linear_combination((on_basis(index), coeff)
+                                         for (index, coeff) in mc.items())
```
Comment (mantepse) [inline src/sage/rings/polynomial/integer_valued_polynomials.py]: personally, I prefer `for index, coeff in mc.items()`, and I think that this is more common in sage.

Comment (fchapoton) [reply src/sage/rings/polynomial/integer_valued_polynomials.py]: right, indeed. Done

**Comment by fchapoton** [comment]
good to go, maybe ?

**Review by mantepse — APPROVED** [review]
Looks good.

**Comment by fchapoton** [comment]
Merci, arigato, danke schön, grazie !

**Pull request merged**